include/oneapi/dpl/pstl/hetero/algorithm_ranges_impl_hetero.h

```diff
@@ -415,30 +514,57 @@ __pattern_search(__hetero_tag<_BackendTag> __tag, _ExecutionPolicy&& __exec, _R1
 // search_n
 //------------------------------------------------------------------------
 
+template <typename _Tp>
+struct __pattern_search_n_fn
+{
+    const _Tp& __value;
```

Comment:
@MikeDvorskiy what do you think about this change?
What is the proper way to declare `__value` at this place: as `const _Tp& __value;` or `_Tp __value;` ?